src/scalar.rs

```diff
@@ -25,14 +26,21 @@ use rkyv::{Archive, Deserialize as RkyvDeserialize, Serialize as RkyvSerialize};
 // The internal representation of this type is four 64-bit unsigned
 // integers in little-endian order. `Scalar` values are always in
 // Montgomery form; i.e., Scalar(a) = aR mod q, with R = 2^256.
-#[derive(Clone, Copy, Eq, Hash)]
+#[derive(Clone, Copy, Eq)]
 #[cfg_attr(
     feature = "rkyv-impl",
     derive(Archive, RkyvSerialize, RkyvDeserialize),
     archive_attr(derive(CheckBytes))
 )]
 pub struct Scalar(pub [u64; 4]);
 
+impl Hash for Scalar {
+    #[inline]
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.0.hash(state);
+    }
+}
+
```

Comment:
Since we have separated the upstream code from our additions, I think I would prefer to have the manual implementation of `Hash` also in 'src/scalar/dusk.rs' (maybe together with a comment that explains why we don't derive as in the upstream crate).